### PR TITLE
actions: compliance: Add support for GitHub annotations

### PR DESCRIPTION
**Pull request by carlescufi**
Add support for annotations as described:
https://docs.github.com/en/actions/using-workflows/workflow-commands-for-github-actions#about-workflow-commands

Note: DNM because I will be using this PR to test it.

### CODE REVIEW

File: scripts/ci/check_compliance.py [hunk 1]
```diff
@@ -990,6 +1033,18 @@ def inheritors(klass):
     return subclasses
 
 
+def annotate(res):
+    """
+    https://docs.github.com/en/actions/using-workflows/workflow-commands-for-github-actions#about-workflow-commands
+    """
+    if not isinstance(res, FmtdFailure):
+        return
```
Comment (gmarull) [inline scripts/ci/check_compliance.py:1041]: is this check needed?

Comment (carlescufi) [reply scripts/ci/check_compliance.py:1041]: nope, will remove, it's a leftover, thanks.

**Comment by galak** [comment]
What does annotations get us?

**Comment by nashif** [comment]
> What does annotations get us?

this: 
image


You do not have to look at the action log and figure out what the text output is talking about, the comments are directly placed where the violation happens.

**Review by stephanosio — APPROVED** [review]

**Review by nashif — APPROVED** [review]

**Pull request merged**